src/utils/conversions.rs

```diff
@@ -1,180 +1,228 @@
+            TemperatureBuilder::from_str("123 f")
+                .unwrap()
+                .as_fahrenheit(),
         );
     }
 
     #[test]
     fn celsius_from_string() {
-        assert!(
-            123.0
-                - TemperatureBuilder::from_str("123C")
-                    .unwrap()
-                    .as_celsius()
-                    .round()
-                < f64::EPSILON
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123C").unwrap().as_celsius(),
+        );
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123 C").unwrap().as_celsius(),
+        );
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123 c").unwrap().as_celsius(),
         );
     }
 
     #[test]
     fn kelvin_from_string() {
-        assert!(
-            123.0
-                - TemperatureBuilder::from_str("123K")
-                    .unwrap()
-                    .as_kelvin()
-                    .round()
-                < f64::EPSILON
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123K").unwrap().as_kelvin(),
+        );
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123 K").unwrap().as_kelvin(),
+        );
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123 k").unwrap().as_kelvin(),
         );
     }
 
     #[test]
     fn rankine_from_string() {
-        assert!(
-            123.0
-                - TemperatureBuilder::from_str("123R")
-                    .unwrap()
-                    .as_rankine()
-                    .round()
-                < f64::EPSILON
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123R").unwrap().as_rankine(),
+        );
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123 R").unwrap().as_rankine(),
+        );
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123 r").unwrap().as_rankine(),
         );
     }
 
     #[test]
     fn default_from_string() {
-        assert!(
-            123.0
-                - TemperatureBuilder::from_str("123")
-                    .unwrap()
-                    .as_celsius()
-                    .round()
-                < f64::EPSILON
+        assert_almost_equal(
+            123.0,
+            TemperatureBuilder::from_str("123").unwrap().as_celsius(),
         );
     }
 
     #[test]
     fn zero_from_string() {
-        assert!(
-            TemperatureBuilder::from_str("")
-                .unwrap()
-                .as_celsius()
-                .round()
-                - 0.0
-                < f64::EPSILON
-        );
+        assert_eq!(0.0, TemperatureBuilder::from_str("").unwrap().as_celsius());
+        assert_eq!(0.0, VolumeBuilder::from_str("").unwrap().as_litres());
```

Comment:
clippy complains about these, I guess can use assert_almost_equal here too.

Reply:
No, cannot use to compare against zero.